docs/plain-bundle-spec.md

```diff
@@ -0,0 +1,151 @@
+
+> Note: there must be at least one resource in the manifests directory in order for the bundle to be a valid
+> plain+v0 bundle.
+
+## Technical Details
+
+* The static manifests must be located in the root-level /manifests directory in a bundle image for the bundle to be a
+  valid `plain+v0` bundle that the provisioner can unpack. A plain bundle image without a /manifests directory is
+  invalid and will not be successfully unpacked onto the cluster.
+* For a bundle git repo, this limitation does not exist, and the manifests can be in any directory in the repository.
+  The manifest directory is assumed to be ./manifests in a bundle git repo but can be provided at runtime.
+* The manifests directory should be flat: all manifests should be at the top-level with no subdirectories.
+* The plain bundle image can be built from any base image, but `scratch` is recommended as it keeps the resulting bundle
+  image a minimal size.
+* Including any content in the plain bundle that is not static manifests will result in a failure when creating content
+  on-cluster from that bundle via a [BundleInstance](/README.md#BundleInstance). Essentially, any file that would not
+  successfully `kubectl apply` will result in an error, but multi-object YAML files, or JSON files, are fine. There will
+  be validation tooling provided that can determine whether a given artifact is a valid bundle.
+
+## Quickstart
+
+As an example, we can package the plain provisioner into a `plain+v0` bundle image by taking the following steps:
+
+1. Let's start by taking a look at the provisioner manifests directory to make sure it is a valid bundle.
+
+```bash
+$ tree ./internal/provisioner/plain/manifests
+./internal/provisioner/plain/manifests
+├── 00_namespace.yaml
+├── 01_cluster_role.yaml
+├── 02_role.yaml
+├── 03_serviceaccount.yaml
+├── 04_cluster_role_binding.yaml
+├── 05_role_binding.yaml
+└── 06_deployment.yaml
+```
+
+This manifests directory is a flattened directory that contains arbitrary Kubernetes manifests and therefore can be
+sourced and unpacked by the plain provisioner. Let's package it up as a plain bundle image.
+
+2. Create a new Dockerfile at the root of the RukPak repository named Dockerfile.example
+
+```bash
+$ touch Dockerfile.example
+```
+
+3. Edit the Dockerfile to include the following:
+
+```bash
+$ cat <<EOF > Dockerfile.example
+FROM scratch
+COPY /internal/provisioner/plain/manifests /manifests
+EOF
+```
+
+4. Build the image using a container tool like docker or podman. Use an image tag that references a repository that you
+   have push access to. For example,
+
+```bash
+$ docker build -f Dockerfile.example -t quay.io/operator-framework/plain-provisioner:example .
+```
+
+5. Push the image to the remote registry
+
+```bash
+$ docker push quay.io/operator-framework/plain-provisioner:example
+```
+
+6. Make sure rukpak is installed locally on a running cluster.
+
+```bash
+$ make run
```

Comment:
Nvmd: I think I see what's happening. We're creating a plain manifest bundle for the provisioner, then we're installing the provisioner separately with `make run`, then we're creating a `Bundle` so that we can see the contents of the provisioner's manifests.

It's pretty confusing IMO to use the plain provisioner's manifests for the bundle image.

Reply:
That's true, I partially chose it because it meant the user didn't have to go and pull down another repository or set of files in order to build a bundle.

We could potentially use combo but it's `manifests` directory is not flat and so it doesn't adhere to the spec. 

I made https://github.com/exdx/combo-bundle for testing purposes, we could point to that instead, WDYT?

Reply:
Can we simply update combo and deflate everything to a single directory?